ha/k8s_setup/ha_setup.py

```diff
@@ -370,15 +372,26 @@ def _add_cluster_component_health(self) -> None:
                                    specific_info=specific_info)
 
 
-    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids) -> None:
+    def _add_node_health(self, data_node_ids, server_node_ids, control_node_ids, kv_enable_batch_put: bool = True) -> None:
         """
         Add node health
         """
         _, _, node_mapping = self._confStoreAPI.get_cluster_cardinality()
+        node_id_list = []
+        node_id_len = len(node_mapping)
         # cluster cardinality node_to_name mapping will have actual machine-ids
         # If that is part of one of the list from data, server or control node
         # list, then mark the respective functional type and update the specific info
         for node_id in node_mapping.values():
+            node_id_list.append(node_id)
+            if len(node_id_list) > CONSUL_TRANSACTIONS_LIMIT:
```

Comment:
Instead of calculating the length of the list in every iteration, just maintain a separate counter and check that,
e.g.
```
nr_transactions = 0
for node_id in node_mapping.values():
    if nr_transactions < CONSUL_TRANSACTIONS_LIMIT:
        nr_transactions += 1
        continue
    nr_transactions = 0   
    if kv_enabl_batch_put:
        ...
```
or if you need node_id list, then you can use list comprehension directly, you already are calculating length node_mapping, so
e.g.
```
node_ids = node_mapping.values()
start = 0
while start < node_id_len:
    node_id_list_to_commit = node_ids[start: start + CONSUL_TRANSACTIONS_LIMIT]
    # commit
    start += CONSUL_TRANSACTIONS_LIMIT
```

Reply:
Instead of this, I am using enumeration now. Which will give an index also. And the same index can be used to track the number of transactions that are already part of batch put.